src/Utilities/Compiler/SmallDictionary.cs

```diff
@@ -571,7 +574,7 @@ private void PushIfNotNull(AvlNode? child)
                 {
                     if (child != null)
                     {
-                        _stack.Push(child);
+                        _stack?.Push(child);
```

Comment:
💡 Rather than use `?.`, you can move this to be a local function of the `MoveNext()` method, and the NRT analysis will see that it is only called when `_stack` is non-null.

This applies to all three cases from the PR.

Reply:
I have done the local function but with `struct` I can't access fields from local func so I had to pass it along.